### PR TITLE
[TE][subscription] update subscription watermarks to use anomaly create time instead of end time

### PR DESCRIPTION
**Pull request by akshayrai**
Problem Statement: 
The current subscription watermarks are designed to notify an anomaly only once (even if merged) and we maintain this by keeping track of the last notified anomaly end time (watermark). However, the assumption here was that newer anomalies will always be detected on newer data (that is, newer anomalies can never have start time < watermark). This puts the restriction when dealing with backfilled data and also in the case of missing data where the actual deviation anomalies on the data might be detected later. This PR tries to remove this restriction by leveraging the anomaly create time in the watermark.

Proposed changes:
* Replace the anomaly end time with the anomaly create time in the vector clock.
* Remove highWaterMark field from subscription config - As of today, we maintain 2 watermarks, namely the last notified anomaly ID(highWaterMark) and the anomaly end time watermark(vector clocks) to ensure that each anomaly is notified only once. The main purpose of the highWaterMark is to filter out merged anomalies from the time window. This is no longer required once we start relying on the anomaly create time.

### CODE REVIEW

File: thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/detection/validators/SubscriptionConfigValidator.java [hunk 1]
```diff
@@ -90,10 +90,6 @@ public void validateConfig(DetectionAlertConfigDTO alertConfig) throws IllegalAr
    */
   @Override
   public void validateYaml(Map<String, Object> config) throws IllegalArgumentException {
-    // Make sure recipients are configured at the right place
-    Preconditions.checkArgument(!config.containsKey(PROP_RECIPIENTS),
```
Comment (jihaozh) [inline thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/detection/validators/SubscriptionConfigValidator.java]: Are we removing this? Seems unrelated.

Comment (akshayrai) [reply thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/detection/validators/SubscriptionConfigValidator.java]: That's already remove in another PR and merged. I need to rebase the changes.

**Comment by xiaohui-sun** [comment]
Thanks for this fix. 
Can you also update needNotification() in DetectionAlertJob class and use the create time?
